src/EFCore.MySql/Query/Internal/MySqlDateTimeMethodTranslator.cs

```diff
@@ -59,7 +63,11 @@ public MySqlDateTimeMethodTranslator(ISqlExpressionFactory sqlExpressionFactory)
                                 new SqlExpression[]
                                 {
                                     _sqlExpressionFactory.Fragment("INTERVAL"),
-                                    _sqlExpressionFactory.Convert(arguments[0], typeof(int)),
+                                    datePart.Equals("microsecond")
+                                        ? _sqlExpressionFactory.Multiply(
+                                            _sqlExpressionFactory.Constant(1000, new IntTypeMapping("int32", DbType.Int32)),
```

Comment:
Is the explicit type mapping really necessary? I would have assumed, that it works just fine without it.